qemu/tests/blockdev_inc_backup_xpt_incon_bitmap.py

```diff
@@ -0,0 +1,159 @@
+import json
+import logging
+import time
+import string
+from avocado.utils import process
+
+from provider import backup_utils
+from provider.blockdev_base import BlockdevBaseTest
+from provider.nbd_image_export import QemuNBDExportImage
+
+from virttest import utils_misc
+
+
+class BlkdevIncXptInconBitmap(BlockdevBaseTest):
+
+    def __init__(self, test, params, env):
+        super(BlkdevIncXptInconBitmap, self).__init__(test, params, env)
+        self.source_images = []
+        self.full_backups = []
+        self.bitmaps = []
+        self.nbd_exports = []
+        self.src_img_tags = params.objects("source_images")
+        list(map(self._init_arguments_by_params, self.src_img_tags))
+
+    def _init_arguments_by_params(self, tag):
+        image_params = self.params.object_params(tag)
+        image_chain = image_params.objects("image_backup_chain")
+        self.source_images.append("drive_%s" % tag)
+        self.full_backups.append("drive_%s" % image_chain[0])
+        self.bitmaps.append("bitmap_%s" % tag)
+        image_params["nbd_export_bitmaps"] = "bitmap_%s" % tag
+        self.nbd_exports.append(QemuNBDExportImage(image_params, tag))
+
+    def do_full_backup(self):
+        extra_options = {"sync": "full",
+                         "persistent": True,
+                         "auto_disable_bitmap": False}
+        backup_utils.blockdev_batch_backup(
+            self.main_vm,
+            self.source_images,
+            self.full_backups,
+            list(self.bitmaps),
+            **extra_options)
+
+    def prepare_data_disk(self, tag):
+        """
+        Override this function, only make fs and mount it
+        :param tag: image tag
+        """
+        self.format_data_disk(tag)
+
+    def gen_inc_files(self):
+        return list(map(self.generate_data_file, self.src_img_tags))
+
+    def kill_vm_after_restart(self):
+        logging.info("Re-start vm again")
+        self.main_vm.create()
+        time.sleep(20)
```

Comment:
if we have to wait till guest started, we may try to log into the vm here;

Reply:
Have updated, thanks.